dtrace-parser/src/dtrace.pest

```diff
@@ -17,8 +17,10 @@ IDENTIFIER = @{ ASCII_ALPHA+ ~ (ASCII_ALPHANUMERIC | "_")* }
 BIT_WIDTH = @{ "8" | "16" | "32" | "64" }
 SIGNED_INT = ${ "int" ~ BIT_WIDTH ~ "_t" }
 UNSIGNED_INT = ${ "uint" ~ BIT_WIDTH ~ "_t" }
-STRING = { "char" ~ "*" }
-DATA_TYPE = { STRING | UNSIGNED_INT | SIGNED_INT}
+STAR = ${ "*" }
+INTEGER = ${ ((SIGNED_INT | UNSIGNED_INT) ~ STAR) | (SIGNED_INT | UNSIGNED_INT) }
```

Comment:
or maybe even

```
INTEGER = ${ (SIGNED_INT | UNSIGNED_INT | "char") }
```

Reply:
This was my initial approach. It's probably better, but led me to a lot more code in Rust. That was probably because I tried to represent the pointer-to-int as another enum variant, transliterating the grammar. I'll give it another whirl, though, see if I can't make it simpler.

Reply:
Ok, I've made the data type rule effectively what you've got here, and also have a separate enum in the Rust representation for a pointer-to-int. The pointer-ness of the type was previously a boolean field. Let me know if this seems better!

Reply:
I think it maybe looks better; but I defer to you. I didn't realize how it might impact the representation; I hope I didn't send you on the wrong path..

Reply:
Not at all. I had tried it originally, and flaked out due to laziness I think. I like this better, considering integers and pointers different types.